src/browser/wisp_network.js

```diff
@@ -0,0 +1,571 @@
+                    // console.log("Sending back data: ");
+                    // console.log(data);
+                    deref.write(data);
+                },
+                closeCallback: (data) => {
+                    deref.close();
+                }
+            });
+            deref.accept(packet);
+
+            return;
+        }
+
+        if(!this.tcp_conn[tuple]) {
+            dbg_log(`I dont know about ${tuple}, so restting`, LOG_FETCH);
+            let bop = packet.tcp.ackn;
+            if(packet.tcp.fin || packet.tcp.syn) bop += 1;
+            reply.tcp = {
+                sport: packet.tcp.dport,
+                dport: packet.tcp.sport,
+                seq: bop,
+                ackn: packet.tcp.seq + (packet.tcp.syn ? 1: 0),
+                winsize: packet.tcp.winsize,
+                rst: true,
+                ack: packet.tcp.syn
+            };
+            this.receive(make_packet(reply));
+            return;
+        }
+
+        this.tcp_conn[tuple].process(packet);
+    }
+
+    if(packet.arp && packet.arp.oper === 1 && packet.arp.ptype === ETHERTYPE_IPV4) {
+        if(!this.masquerade) {
+            let packet_subnet = iptolong(packet.arp.tpa) & 0xFFFFFF00;
+            let router_subnet = iptolong(this.router_ip) & 0xFFFFFF00;
+
+            if(packet_subnet !== router_subnet) {
+                return;
+            }
+        }
+
+        // Reply to ARP Whohas
+        let reply = {};
+        reply.eth = { ethertype: ETHERTYPE_ARP, src: this.router_mac, dest: packet.eth.src };
+        reply.arp = {
+            htype: 1,
+            ptype: ETHERTYPE_IPV4,
+            oper: 2,
+            sha: this.router_mac,
+            spa: packet.arp.tpa,
+            tha: packet.eth.src,
+            tpa: packet.arp.spa
+        };
+        this.receive(make_packet(reply));
+
+    }
+
+    if(packet.dns) {
+        (async () => {
+            let reply = {};
+            reply.eth = { ethertype: ETHERTYPE_IPV4, src: this.router_mac, dest: packet.eth.src };
+            reply.ipv4 = {
+                proto: IPV4_PROTO_UDP,
+                src: this.router_ip,
+                dest: packet.ipv4.src,
+            };
+            reply.udp = { sport: 53, dport: packet.udp.sport };
+
+            let answers = [];
+            let flags = 0x8000; //Response,
+            flags |= 0x0180; // Recursion
+            // flags |= 0x0400; Authoritative
+            for(let i = 0; i < packet.dns.questions.length; ++i) {
+                let q = packet.dns.questions[i];
+                // Sometimes this results in no answer at all, like if dohdns crashes, but UDP is unreliable in nature so this shouldn't be that big of an issue
+                const res = await dohdns(q);
+                switch(q.type){
+                    case 1: // A recrod
+
+                        // for (const ans in res.Answer) {    // v86 DNS server crashes and burns with multiple answers, not quite sure why
```

Comment:
> v86 DNS server crashes and burns with multiple answers, not quite sure why

What dns server does this refer to? The one in the alpine profile?

I think it's more likely a bug in make_packet

Reply:
Was referring to the internal DNS server in fetch_networking.js, if I pushed all the answers it wasn't very happy, it most likely is a make_packet bug I agree